File: 21-merge-two-sorted-lists/merge-two-sorted-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(list1, list2) {
    let arr = [];

    while(!(list1 == undefined)){
        arr.push(list1.val);
        list1 = list1.next;
    }

    while(!(list2 == undefined)){
        arr.push(list2.val);
        list2 = list2.next;
    }

    arr.sort((a,b) => a-b);
    if(arr.length ==0) return list1;

    

    let res=  new ListNode(arr[0], undefined);
    let tmp = res;

    for(let i = 1;i<arr.length;i++){
        tmp.next = new ListNode(arr[i], undefined);
        tmp = tmp.next;
    }

    return res;
};

File: 21-merge-two-sorted-lists/merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:

        tmp = ListNode()
        res = tmp

        if list1 != None and list2 != None:
            if list1.val<list2.val :
                tmp.val = list1.val
                list1 = list1.next
            else:
                tmp.val = list2.val
                list2 = list2.next
        elif list1 != None:
            tmp.val = list1.val
            list1 = list1.next
        elif list2 != None:
            tmp.val = list2.val
            list2 = list2.next
        else:
            res = None
        
        if list1 != None or list2 != None:
                tmp.next = ListNode()
                tmp = tmp.next
        while list1 != None or list2 != None:
            if list1 != None and list2 != None:
                if list1.val<list2.val :
                    tmp.val = list1.val
                    list1 = list1.next
                else:
                    tmp.val = list2.val
                    list2 = list2.next
            elif list1 != None:
                tmp.val = list1.val
                list1 = list1.next
            else:
                tmp.val = list2.val
                list2 = list2.next
            if list1 != None or list2 != None:
                tmp.next = ListNode()
                tmp = tmp.next
        return res
        